services/QuillLMS/app/services/units/updater.rb

```diff
@@ -11,16 +11,13 @@ def self.run(unit_id, activities_data, classrooms_data, current_user_id=nil)
   end
 
   def self.assign_unit_template_to_one_class(unit_id, classrooms_data, unit_template_id, current_user_id=nil, concatenate_existing_student_ids: false)
-    classroom_array = [classrooms_data]
-    # converted to array so we can map in helper function as we would otherwise
-    unit_template = UnitTemplate.find(unit_template_id)
-    activities_data = unit_template.activities.map{ |a| {id: a.id, due_date: nil} }
-    update_helper(unit_id, activities_data, classroom_array, current_user_id, concatenate_existing_student_ids: concatenate_existing_student_ids)
+    activities_data = UnitTemplate.find(unit_template_id).activities.map { |a| {id: a.id, due_date: nil} }
+    update_helper(unit_id, activities_data, [classrooms_data], current_user_id, concatenate_existing_student_ids: concatenate_existing_student_ids)
   end
 
   def self.fast_assign_unit_template(teacher_id, unit_template, unit_id, current_user_id=nil)
-    activities_data = unit_template.activities.select('activities.id AS id, NULL as due_date')
-    classrooms_data = User.find(teacher_id).classrooms_i_teach.map{|classroom| {id: classroom.id, student_ids: [], assign_on_join: true}}
+    activities_data = unit_template.activities.pluck(:id).map { |activity_id| { id: activity_id, due_date: nil } }
```

Comment:
Much prefer doing this in Ruby than with a weird SELECT in SQL.  Thanks for the clean-up.